pootle/apps/accounts/utils.py

```diff
@@ -129,6 +130,12 @@ def merge_submissions(self):
         """
         # Update submitter on submissions
         for submission in self.src_user.submission_set.iterator():
+
+            if not submission.unit:
+                # This appears to be an orphaned submission
+                submission.delete()
+                continue
```

Comment:
Is it not more efficient to bulk delete these? I mean instead of deleting each of these submissions inside this `for` loop, do bulk delete them before getting into the loop:

```python
        # Delete orphaned submissions.
        self.src_user.submission_set.filter(unit__isnull=True).delete()

        for submission in self.src_user.submission_set.iterator():
```

The same applies to the next change below.